Why Python is Essential for Data Analysis

majiezra opened the pull request:
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [X] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [X] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [X] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

this fixes  #1934

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+### Introduction
+
+Data is an essential part of any enterprise/business.
+It is critical to collect, process, and analyze data flow in a timely and accurate manner.
+ 
+The data science industry is rapidly expanding. Data volume can be large, making information handling difficult and time-consuming.
```
```suggestion
Data is an essential part of any enterprise/business. It is critical to collect, process, and analyze data flow in a timely and accurate manner.
 
The data science industry is rapidly expanding. Data volume can be large, making information handling difficult and time-consuming.
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+ 
+The data science industry is rapidly expanding. Data volume can be large, making information handling difficult and time-consuming.
+
+Python is a popular programming language in scientific computing because it’s many data-oriented feature packages can speed up and simplify data processing, saving time.
```
```suggestion
Python is a popular programming language in scientific computing because its many data-oriented feature packages can speed up and simplify data processing, thus saving time.
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+### What is Python?
+
+Python is an object-oriented, high-level programming language.
+Its built-in data structures and properties, combined with dynamic typing and binding, make it ideal for Application Development and its use as a scripting language. Python's simple syntax emphasizes readability and reduces programming maintenance.
```
```suggestion
Python is an object-oriented, high-level programming language.

Its built-in data structures and properties, combined with dynamic typing and binding, make it ideal for Application Development and its use as a scripting language. Python's simple syntax emphasizes readability and reduces programming maintenance.
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+See the features below.
+
+### Easy to Learn
```
```suggestion
#### Easy to Learn
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+Python focuses on both simplicity and readability, while also providing a plethora of useful options for data analysts/scientists. As a result, even novices can easily use its relatively simple syntax to create effective solutions for complex scenarios with just fewer lines of code.
+
+### Flexibility
```
```suggestion
#### Flexibility
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+Python's extreme versatility is another powerful attribute that makes it popular among data scientists and analysts. As a result, data models can be created, data sets can be systematized, ML-powered algorithms can be developed, web services can be developed, and data mining can be used to complete various tasks in a short amount of time.
+
+### Huge Libraries Collection
```
```suggestion
#### Huge Libraries Collection
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+It is worth noting that the libraries are constantly expanding, providing robust solutions.
+
+### Graphics and Visualization
```
```suggestion
#### Graphics and Visualization
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+Python provides users with a plethora of different visualization options. As a consequence, it is a must-have method for all data science, not just data processing. By developing numerous charts and graphics, as well as web-ready interactive plots, data analysts can make data more available.
+
+### Built-in data analytics tools
```
```suggestion
#### Built-in data analytics tools
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+The data science industry is rapidly expanding. Data volume can be large, making information handling difficult and time-consuming.
+
+Python is a popular programming language in scientific computing because it’s many data-oriented feature packages can speed up and simplify data processing, saving time.
+
```
It would be great, if you could give a 1 - 2 lines intro on what this article is about, and the key takeaways

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+Data analysis is the process of gathering raw data and converting it into information that users can use to make decisions.
+
+It entails inspecting, cleansing, transforming, and modeling data in order to uncover valuable information, draw conclusions, and aid decision-making.
```
```suggestion
It entails inspecting, cleansing, transforming, and modeling data to uncover valuable information, draw conclusions, and aid decision-making.
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+### What makes python a brilliant choice for Data analysis?
+
+Python is a multi-functional, maximally interpreted programming language with a number of advantages that is often used to streamline massive, complex data sets. Python has a number of distinguishing characteristics that make it the best option for data analysis.
```
```suggestion
Python is a multi-functional, maximally interpreted programming language with several advantages that are often used to streamline massive, complex data sets. Python has a number of distinguishing characteristics that make it the best option for data analysis.
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+### Conclusion
+
+Any company's ability to derive information and insights from data in order to make good strategic decisions, stay competitive, and progress is directly linked to its performance. Python is an internationally recognized programming language that can assist users in better managing of data for a variety of reasons.
```
```suggestion
Any company's ability to derive information and insights from data in order to make good strategic decisions, stay competitive, and progress is directly linked to its performance. Python is an internationally recognized programming language that can assist users in better managing data for a variety of reasons.
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+- What is Data analysis?
+- Who is a Data analyst?
+- What makes Python a brilliant choice for Data analysis?
+- Conclusion
```
```suggestion
- Conclusion
- Reference
```

srishilesh commented on articles/why-python-is-essential-for-data-analysis/index.md:
```diff
@@ -0,0 +1,84 @@
+
+### Reference
+
+[pdfdrive](https://www.pdfdrive.com/python-data-analytics-data-analysis-and-science-using-pandas-matplotlib-and-the-python-programming-language-e158003322.html)
```
```suggestion
- [pdfdrive](https://www.pdfdrive.com/python-data-analytics-data-analysis-and-science-using-pandas-matplotlib-and-the-python-programming-language-e158003322.html)
```

srishilesh replied:
This link doesn't seem to work. It shows 404 for me. Could you add some other link?

srishilesh reviewed (changes requested):
Hey @majiezra,

Thanks for the PR.

Great article. I have suggested very minor changes.

Please let me know when they're done, we can move on to the tech review.

Thank you.

srishilesh commented:
@majiezra 

And, you must have a hero image of dimensions 600x400.

I believe `datascience.png` is a hero image. If so, please rename it.

majiezra commented:
> Hey @majiezra,
> 
> Thanks for the PR.
> 
> Great article. I have suggested very minor changes.
> 
> Please let me know when they're done, we can move on to the tech review.
> 
> Thank you.

majiezra commented:
> @majiezra
> 
> And, you must have a hero image of dimensions 600x400.
> 
> I believe `datascience.png` is a hero image. If so, please rename it.

Thank you very much for the observations. I have implemented the minor changes and the reference link works well because I just tried it. Thanks. I have changed the hero image to `jpg`

majiezra commented:
@srishilesh I have made the changes. Thank you

srishilesh commented:
> > @majiezra
> > 
> > And, you must have a hero image of dimensions 600x400.
> > 
> > I believe `datascience.png` is a hero image. If so, please rename it.
> 
> Thank you very much for the observations. I have implemented the minor changes and the reference link works well because I just tried it. Thanks. I have changed the hero image to `jpg`

Hey, 

Thanks for the changes. Will do the next round of review by today.

I meant to rename the hero image as "hero.jpg".

srishilesh reviewed (approved):
Hey @majiezra,

Awesome article. It was beginner friendly. Keep up the great work!

Please make sure to go through my changes carefully, so that you could improve on your future articles.

Also, from the next time, make sure to add a `hero` image like `hero.jpg` (can be `png` too) of dimensions `600x400`.

Moving this to the final review.

Thank you.

majiezra commented:
@srishilesh thank you.

hectorkambow commented:
plagiarism check done
